### PR TITLE
Refactor: Redesign media list on Eligibility Start page

### LINKED ISSUE #365: Refactor rendering of "media list" on Eligibility Start
## Background

The [`eligibility:start` view](https://github.com/cal-itp/benefits/blob/dev/benefits/eligibility/views.py#L58) and [`core/page.html` template](https://github.com/cal-itp/benefits/blob/dev/benefits/core/templates/core/page.html) are responsible for rendering the "Get Started" page.

The view [sends a list of `MediaItem`](https://github.com/cal-itp/benefits/blob/dev/benefits/eligibility/views.py#L67) that the page template renders via the includes [`core/includes/media-list.html`](https://github.com/cal-itp/benefits/blob/dev/benefits/core/templates/core/includes/media-list.html) and [`core/includes/media-item.html`](https://github.com/cal-itp/benefits/blob/dev/benefits/core/templates/core/includes/media-item.html)

### New view - Desktop

image

### New view - Mobile (rough sketch)

image

### Where to render

The Eligibility Start page is the only page that renders a list of media items like this. 

* [x] Move rendering of media list from the generic page template/viewmodel to the new template from #374 

### What items to render

* [ ] If SSO is enabled (#372), render the SSO item first
* [ ] Always render the `verifier`'s item second to last
* [ ] Always render the payment card item last

### What to render for each item:

* [x] Icon + alt text
* [x] Title
* [x] Description
* [ ] 0 or more link button(s)

### How to render

The CA State Template has a component called ["Progress Blocks"](https://template.webstandards.ca.gov/sample/progress-blocks.html) that may be useful here

We want to use the new treatment (full-width, dashed lines connection) for all flows; in other words, this is the style the Start page should render for Courtesy Cards (with only 2 items, since Login.gov would not be required).

We also need to look at expanding the Start page template to be a more full-width view in Desktop. The current container constrains the page more than what this Mock calls for.

### LINKED ISSUE #408: Figure out why Login.gov button tests don't pass in GitHub
See #402 

E.g. in run [`5841870888`](https://github.com/cal-itp/benefits/runs/5841870888?check_suite_focus=true) the "Contains Login.gov button" test failed.

Eventually in [`29fb047`](https://github.com/cal-itp/benefits/pull/402/commits/29fb047b64327c7d1cee2a2ea39c2d0165ecc3da) we skipped the failing test to unblock the merge.

Yet this test (and other UI tests) pass when running locally, in the devcontainer:

image

image

And in the Cypress runner:

image

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #365.
Closes #408.

Building on the template from #391, this PR moves the rendering of the media list that was previously handled in the base Page template, to the new Start page template. This is the only page that renders items like this, so it does not need to be generic.

### User stories
- If there's authentication required, the page will show the authentication step instructions. If not, it will not show the authentication step instructions.
- Users will have Help links (opens in same tab) and Login.gov links (opens in new tab)

### Templates

1. Removes `media-list.html` template from `page` template, creates and uses a new `start` page template just for this page.

### Tests
1. Adds an `upload-artifact` step to the Tests UI workflow. This step will upload Cypress videos to GitHub, along with the screenshots, when a UI test fails. This video is useful for better debugging broken tests.
2. Separates Eligibility specs into 2 files: One for the Authentication flow, and one for the flow without Authentication.
3. Adds more granular UI specs for the Authentication flow

### Login.gov flow

image
image
image

### Non-Login.gov flow

image
image

### Spanish

image
image

### CODE REVIEW

**Comment by thekaveman** [comment]
I put this back in draft while you're working on the tests... yeah I'm still not sure what's up with the Login.gov button test?

#408

File: benefits/eligibility/views.py [hunk 1]
```diff
@@ -63,30 +63,67 @@ def start(request):
...
+        dict(
+            icon=viewmodels.Icon("bankcardcheck", pgettext("image alt text", "core.icons.bankcardcheck")),
+            heading=_("eligibility.pages.start.items[1].title"),
+            details=_("eligibility.pages.start.items[1].text"),
+            links=[
+                viewmodels.Button.link(
+                    classes="btn-text btn-link",
+                    text=_("eligibility.pages.start.items[1].button[0].link"),
+                    url=_("eligibility.pages.start.items[1].button[0].url"),
+                    target="_blank",
```
[the remark this answers is not in the document]
Comment (machikoyasuda) [reply benefits/eligibility/views.py:111]: Yeah they go to the /help page. The slightly unfortunate part is that when the user is on the Help page, and clicks the `Go back` button, they will be taken back to the page with the radio buttons page, not the instructions page. That seemed like a subpar experience so I set these to open in a new tab.

[the remark this answers is not in the document]
Comment (thekaveman) [reply benefits/eligibility/views.py]: This is what  the [`session.origin()`](https://github.com/cal-itp/benefits/blob/dev/benefits/core/session.py#L126) is all about - saving the "last" page the user was on for those Back buttons.

In #414 ([commit `396a874`](https://github.com/cal-itp/benefits/pull/414/commits/396a874dff94ca44cb676d33e2035f2b20d6f43f)) I updated origin to `eligibility:start` in this view, you could do the same here to test that the back button is correctly updated - we can deal with a potential merge conflict later.

[the remark this answers is not in the document]
Comment (machikoyasuda) [reply benefits/eligibility/views.py]: Fixed!! 
image

**Comment by machikoyasuda** [comment]
Latest changes from dev have been merged in 🙏

**Review by thekaveman — COMMENTED** [review]
I approve! ✅ 

Just can't "Approve" officially since I opened the PR original 🙄

**Review by machikoyasuda — APPROVED** [review]
Approving on behalf of @thekaveman

**Pull request merged**